### PR TITLE
[MonoError] Introduce fast (and short) macros for initing and checking MonoError. Optimize its size.

### PR DESCRIPTION
**Pull request by kumpera**

### CODE REVIEW

**Comment by kumpera** [comment]
@vargaz I believe this will be backwards compat, right?

**Comment by vargaz** [comment]
It will be source compatible.

File: mono/utils/mono-error-internals.h [hunk 1]
```diff
@@ -18,11 +18,17 @@ typedef struct {
...
+	const char *first_argument;
 
-	void *padding [4];
-    char message [128];
+	void *padding [3];
 } MonoErrorInternal;
 
+#define error_init(error) do {	\
+	(error)->error_code = MONO_ERROR_NONE;	\
+} while (0);
```
Comment (ludovic-henry) [inline mono/utils/mono-error-internals.h]: this doesn't do the same thing as `mono_error_init`, as it doesn't initialize `(error)->flags = 0`

**Comment by ludovic-henry** [comment]
LGTM

**Comment by kumpera** [comment]
Let me fix that macro then we merge

**Pull request merged**